specification/schemaregistry/data-plane/Microsoft.EventHub/preview/2018-01-01-preview/schemaregistry.json

```diff
@@ -29,104 +29,300 @@
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "headers": {
+              "Location": {
+                "type": "string",
+                "description": "URL location of schema, identified by schema group, schema name, and version."
+              },
+              "X-Schema-Type": {
+                "type": "string",
+                "description": "Serialization type for the schema being stored."
+              },
+              "X-Schema-Id": {
+                "type": "string",
+                "description": "References specific schema in registry namespace."
+              },
+              "X-Schema-Id-Location": {
+                "type": "string",
+                "description": "URL location of schema, identified by schema ID."
+              },
+              "X-Schema-Version": {
+                "type": "integer",
+                "description": "Version of the returned schema."
+              }
+            },
+            "schema": {
+              "$ref": "#/definitions/SchemaId"
+            }
+          },
+          "default": {
+            "description": "An error response received from the Azure Schema Registry service.",
+            "schema": {
+              "$ref": "#/definitions/ServiceErrorResponse"
+            }
+          }
+        }
+      },
+      "put": {
+        "summary": "Register new schema",
+        "description": "Register new schema. If schema of specified name does not exist in specified group, schema is created at version 1. If schema of specified name exists already in specified group, schema is created at latest version + 1.\n",
         "x-ms-examples": {
-          "BatchAccountDelete": { "$ref": "./examples/OperationGroupGet.json" }
+          "Register schema": {
+            "$ref": "./examples/OperationSchema_Register.json"
+          }
         },
-        "description": "This is a sample get operation, please see guidelines in azure-rest-api-specs repository for more info",
+        "operationId": "Schema_Register",
+        "produces": [
+          "application/json"
+        ],
         "parameters": [
           {
-            "$ref": "#/parameters/SubscriptionIdParameter"
+            "name": "group-name",
+            "in": "path",
+            "description": "Schema group under which schema should be registered.  Group's serialization type should match the serialization type specified in the request.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "name": "schema-name",
+            "in": "path",
+            "description": "Name of schema being registered.",
+            "required": true,
+            "type": "string"
           },
           {
-            "$ref": "#/parameters/ResourceGroupNameParameter"
+            "name": "serialization-type",
```

Comment:
This header name is not accepted by current implementation. It wants x-schema-type. Is the implementation behind or the spec wrong on this? 

Should this be `required: true`? I get 400 without it and Java has a positional param. 

Also consider x-ms-client-name without a leading 'x-; here, autorest is generating properties like `xSomething` instead of `something`. I'm not blocked on this in TS as I'm not exposing the generated models directly currently.

This comment applies equally to same parameter on QueryIdByContent. Is there a reason the parameters are duped instead of $ref'ed.

Reply:
Ah - sorry, I see the issue.  Let me fix this.

We should rename to Serialization-Type in the preview version.